python/hail/methods/misc.py

```diff
@@ -1,8 +1,125 @@
 from decorator import decorator
-from hail.api2 import MatrixTable
-from hail.utils.java import Env, handle_py4j
+from hail.api2 import MatrixTable, Table
+from hail.utils.java import Env, handle_py4j, jarray_to_list, joption
 from hail.typecheck.check import typecheck, strlike
 from hail.expr.expression import *
+from hail.expr.ast import Reference
+
+@handle_py4j
+@typecheck(i=Expression,
+           j=Expression,
+           tie_breaker=nullable(func_spec(2, expr_numeric)))
+def maximal_independent_set(i, j, tie_breaker=None):
+    """Compute a `maximal independent set`_ of vertices in an undirected graph
+    whose edges are given by a two-column table.
+
+    .. _maximal independent set: https://en.wikipedia.org/wiki/Maximal_independent_set
+
+    Examples
+    --------
+
+    Prune individuals from a dataset until no close relationships remain with
+    respect to a PC-Relate measure of kinship.
+
+    >>> pc_rel = methods.pc_relate(dataset, 2, 0.001)
+    >>> rel_pairs = pc_rel.filter(pc_rel['kin'] > 0.125).select('i', 'j')
```

Comment:
I'd change `rel_pairs` to just `pairs` throughout these examples. It's shorter and clear from context that these are the relevant edges for MIS (and `rel_pairs` is inconsistent with `related_` used everywhere else which gets long)